protostar/starknet/cairo1_parser.py

```diff
@@ -0,0 +1,60 @@
+# pylint: disable="protected-access"
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+import json
+
+from starkware.cairo.lang.compiler.program import Program
+from starkware.starkware_utils.marshmallow_dataclass_fields import IntAsHex
+
+
+@dataclass(frozen=True)
+class CairoHintCode:
+    code: str
+    accessible_scopes: list[Any]
+    flow_tracking_data: Any
+
+
+@dataclass(frozen=True)
+class TestCase:
+    name: str
+    offset: int
+
+
+@dataclass(frozen=True)
+class TestSuite:
+    path: Path
+    test_cases: list[TestCase]
+    program: Program
+
+
+def parse_test_suite(path: Path, json_raw: Any) -> TestSuite:
+    json_dict = json.loads(json_raw)
+    prime: int = IntAsHex()._deserialize(json_dict["prime"], None, None)  # pylint
+    data: list[int] = [
+        IntAsHex()._deserialize(v, None, None) for v in json_dict["bytecode"]
+    ]
+    hints: dict[int, list[CairoHintCode]] = {}
+    for h in json_dict["hints"]:
+        codes = [CairoHintCode(str(e), [None], None) for e in h[1]]
+        hints[int(h[0])] = codes
```

Comment:
This piece of code is hard to read.

- this snippet could be a function
- `hints` could be renamed to `offset_to_hint_codes`
- `int` could be replaced with custom type `Offset`